.gitignore

```diff
@@ -1,8 +1,19 @@
-*.iml
 .gradle
 /local.properties
-/.idea/*
 .DS_Store
 **/build/*
 /captures
 .externalNativeBuild
+
+# IntelliJ
+*.iml
+.idea/workspace.xml
+.idea/tasks.xml
+.idea/gradle.xml
+.idea/assetWizardSettings.xml
+.idea/dictionaries
+.idea/libraries
+.idea/misc.xml
+# Android Studio 3 in .gitignore file.
+.idea/caches
+.idea/modules.xml
```

Comment:
Are we trying to explicitly commit some of `.idea/`?

Reply:
We are, I see them.

Reply:
Yeah - please see the conversation here [here](https://github.com/pusher/chatkit-android/pull/170#discussion_r377747617)

Reply:
I suspect this might pick up some cruft in future when studio adds more temp files. gitignore supports an explicit "don't ignore" with line's starting `!`, so it might be better to have

```
.idea/*
!.idea/.name 
!.idea/codeStyles/*
!.idea/inspectionProfiles/*
!.idea/encodings.xml
[more?]
```

Having the default run profiles available seems nice, but things like running individual tests creates loads more, will we have to take care not to commit them? Because that would be a pain (and experience suggests that we will frequently fail)

Reply:
That's a fair suggestion! I picked up the suggestion from the GitHub android specific gitignore so I had expected it not to add any cruft. Did you find it created those individual files when you ran the tests?

Reply:
I have done a quick check as I had the same concern and nothing has been added to that file when I added individual test run configurations.

Reply:
Excellent :) I guess it keeps the extra run-configs elsewhere

Reply:
> I picked up the suggestion from the GitHub android specific gitignore so I had expected it not to add any cruft.

This seems reasonable, we can switch to whitelisting if new cruft starts appearing.